type-definitions/most.d.ts

```diff
@@ -1,385 +1,788 @@
+		(...disposables: Array<Disposable<any>>): Disposable<Array<any>>;
+	}
+
+	export const all: DisposeAllFn;
+
+	export function settable<T>(): SettableDisposable<T>;
+
+	export function promised<T>(diposablePromise: Promise<Disposable<T>>): Disposable<T>;
+	export function once<T>(disposable: Disposable<T>): Disposable<T>;
+}
+
+declare module "most/lib/disposable/dispose" {
+	export * from 'most/src/disposable/dispose';
+}
+
+declare module "most/src/sink/DeferredSink" {
+	import { Sink } from 'most';
+
+	class DeferredSink<T> implements Sink<T> {
+		constructor(sink: Sink<T>);
+		event(time: number, value: T): void;
+		end(time: number, value?: T): void;
+		error(time: number, err: Error): void;
+	}
+
+	export default DeferredSink;
+}
+
+declare module "most/lib/sink/DeferredSink" {
+	import DeferredSink from 'most/src/sink/DeferredSink';
+	export = DeferredSink;
+}
+
+declare module "most/src/sink/IndexSink" {
+	import { Sink } from 'most';
+
+	class IndexSink<T> implements Sink<T> {
+		public index: number;
+		public value: T;
+		public active: boolean;
+		public sink: Sink<IndexSink<T>>;
+		constructor(index: number, sink: Sink<IndexSink<T>>);
+		event(time: number, value: T): void;
+		end(time: number, value?: T): void;
+		error(time: number, err: Error): void;
+	}
+
+	export default IndexSink;
+}
+
+declare module "most/lib/sink/IndexSink" {
+	import IndexSink from 'most/src/sink/IndexSink';
+	export = IndexSink;
+}
+
+declare module "most/src/sink/Pipe" {
+	import { Sink } from 'most'
+
+	class Pipe<T> implements Sink<T> {
+		constructor(sink: Sink<T>);
+		event(time: number, value: T): void;
+		end(time: number, value?: T): void;
+		error(time: number, err: Error): void;
+	}
+
+	export default Pipe;
+}
+
+declare module "most/lib/sink/Pipe" {
+	import Pipe from 'most/src/sink/Pipe';
+	export = Pipe;
+}
+
+declare module "most/src/sink/SafeSink" {
+	import { Sink } from 'most';
+
+	class SafeSink<T> implements Sink<T> {
+		public active: boolean;
+		constructor(sink: Sink<T>);
+		event(time: number, value: T): void;
+		end(time: number, value?: T): void;
+		error(time: number, err: Error): void;
+		disable(): Sink<T>;
+	}
+
+	export default SafeSink;
+}
+
+declare module "most/lib/sink/SafeSink" {
+	import SafeSink from 'most/src/sink/SafeSink';
+	export = SafeSink;
+}
+
+declare module "most/src/fatalError" {
+	function fatalError (err: Error): void;
+
+	export default fatalError;
+}
+
+declare module "most/lib/fatalError" {
+	import fatalError from 'most/src/fatalError';
+	export = fatalError;
+}
+
+declare module "most/src/invoke" {
+	interface InvokeFn {
+		<A, B>(f: (a: A) => B, args: [A]): B;
+		<A, B, C>(f: (a: A, b: B) => C, args: [A, B]): C;
+		<A, B, C, D>(f: (a: A, b: B, c: C) => D, args: [A, B, C]): D;
+		<A, B, C, D, E>(f: (a: A, b: B, c: C, d: D) => E, args: [A, B, C, D]): E;
```

Comment:
Well it is possible to `n` number of arguments as the `default` case of `invoke` falls back to `fn.apply(...)`